src/common/src/array/data_chunk_iter.rs

```diff
@@ -326,6 +340,16 @@ impl RowDeserializer {
         let datum = deserialize_datum_from(&self.data_types[datum_idx], &mut deserializer)?;
         Ok(datum)
     }
+
+    /// Deserialize the row from a value encoding bytes.
+    pub fn value_decode(&self, mut data: impl Buf) -> RwResult<Row> {
+        let mut values = vec![];
+        values.reserve(self.data_types.len());
+        for ty in &self.data_types {
+            values.push(deserialize_datum(&mut data, ty)?);
```

Comment:
ditto

Reply:
`deserialize_datum` has to returns a `Result`. Can't return `Result` in a map function.